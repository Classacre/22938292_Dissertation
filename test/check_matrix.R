# Check Seurat object RNA assay structure and available slots
library(Seurat)

seurat_path <- "/group/sms029/Oliva_dataset/integrated_col_trajectories.rds"

seurat_obj <- readRDS(seurat_path)

cat("Assay class:", class(seurat_obj@assays$RNA), "\n\n")
cat("Assay slots:\n")
print(slotNames(seurat_obj@assays$RNA))
cat("\nAssay names:\n")
print(names(seurat_obj@assays$RNA))
cat("\nAssay structure:\n")
str(seurat_obj@assays$RNA)
cat("\nAvailable layers (if any):\n")
if ("layers" %in% slotNames(seurat_obj@assays$RNA)) {
  print(names(seurat_obj@assays$RNA@layers))
}
cat("\nSeurat object version:", seurat_obj@version, "\n")
cat("Matrix check complete.\n")
