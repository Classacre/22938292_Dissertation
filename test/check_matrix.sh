#!/bin/bash
#SBATCH --job-name=check_matrix
#SBATCH --output=/group/sms029/mnieuwenh/test/logs/check_matrix.out
#SBATCH --error=/group/sms029/mnieuwenh/test/logs/check_matrix.err
#SBATCH --partition=work
#SBATCH --mem=128G
#SBATCH --cpus-per-task=2
#SBATCH --time=01:00:00
#SBATCH --mail-user=22938292@student.uwa.edu.au
#SBATCH --mail-type=END,FAIL

module load Anaconda3/2024.06
source ~/.bashrc
conda activate /group/sms029/conda_environment/R

Rscript /group/sms029/mnieuwenh/test/check_matrix.R
