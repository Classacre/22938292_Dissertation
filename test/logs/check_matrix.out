Assay class: Assay5 

Assay slots:
[1] "layers"     "cells"      "features"   "default"    "assay.orig"
[6] "meta.data"  "misc"       "key"       

Assay names:
NULL

Assay structure:
Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  ..@ layers    :List of 12
  .. ..$ data        :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. ..@ i       : int [1:241229262] 9 14 73 91 110 135 154 174 189 190 ...
  .. .. .. ..@ p       : int [1:87057] 0 1033 4085 5845 6967 8496 11163 12422 13479 15570 ...
  .. .. .. ..@ Dim     : int [1:2] 25784 87056
  .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. ..$ : NULL
  .. .. .. .. ..$ : NULL
  .. .. .. ..@ x       : num [1:241229262] 2.25 1.66 2.25 2.25 1.66 ...
  .. .. .. ..@ factors : list()
  .. ..$ counts      :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. ..@ i       : int [1:241229262] 9 14 73 91 110 135 154 174 189 190 ...
  .. .. .. ..@ p       : int [1:87057] 0 1033 4085 5845 6967 8496 11163 12422 13479 15570 ...
  .. .. .. ..@ Dim     : int [1:2] 25784 87056
  .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. ..$ : NULL
  .. .. .. .. ..$ : NULL
  .. .. .. ..@ x       : num [1:241229262] 2 1 2 2 1 1 1 2 3 1 ...
  .. .. .. ..@ factors : list()
  .. ..$ scale.data.1: num [1:23592, 1:5313] -0.302 -0.479 -0.226 -0.251 -1.367 ...
  .. ..$ scale.data.2: num [1:23711, 1:7759] -0.306 -0.487 -0.18 -0.268 -1.279 ...
  .. ..$ scale.data.3: num [1:23988, 1:8455] -0.374 -0.449 -0.205 -0.281 0.301 ...
  .. ..$ scale.data.4: num [1:22884, 1:10955] -0.434 -0.439 -0.11 -0.245 0.65 ...
  .. ..$ scale.data.5: num [1:22300, 1:9573] -0.379 -0.388 -0.103 -0.223 -1.216 ...
  .. ..$ scale.data.6: num [1:23541, 1:10911] -0.385 -0.41 -0.125 -0.275 1.351 ...
  .. ..$ scale.data.7: num [1:22959, 1:12031] -0.352 3.981 -0.114 -0.227 1.181 ...
  .. ..$ scale.data.8: num [1:22955, 1:11064] -0.3386 -0.3301 -0.0967 -0.2431 0.8249 ...
  .. ..$ scale.data.9: num [1:23332, 1:10995] -0.307 -0.313 -0.111 -0.24 0.972 ...
  .. ..$ scale.data  : num [1:19634, 1:87056] -0.333 -0.384 -0.128 -0.235 -1.173 ...
  ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. ..@ .Data: logi [1:87056, 1:12] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. ..$ : chr [1:87056] "AAACCTGAGAGCCCAA-1_1" "AAACCTGAGGCCCTTG-1_1" "AAACGGGAGCTAGGCA-1_1" "AAACGGGCACCCTATC-1_1" ...
  .. .. .. .. ..$ : chr [1:12] "scale.data.1" "scale.data.2" "scale.data.3" "scale.data.4" ...
  .. .. ..$ dim     : int [1:2] 87056 12
  .. .. ..$ dimnames:List of 2
  .. .. .. ..$ : chr [1:87056] "AAACCTGAGAGCCCAA-1_1" "AAACCTGAGGCCCTTG-1_1" "AAACGGGAGCTAGGCA-1_1" "AAACGGGCACCCTATC-1_1" ...
  .. .. .. ..$ : chr [1:12] "scale.data.1" "scale.data.2" "scale.data.3" "scale.data.4" ...
  ..@ features  :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. ..@ .Data: logi [1:25784, 1:12] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. ..$ : chr [1:25784] "AT1G01010" "AT1G01020" "AT1G01030" "AT1G01040" ...
  .. .. .. .. ..$ : chr [1:12] "scale.data.1" "scale.data.2" "scale.data.3" "scale.data.4" ...
  .. .. ..$ dim     : int [1:2] 25784 12
  .. .. ..$ dimnames:List of 2
  .. .. .. ..$ : chr [1:25784] "AT1G01010" "AT1G01020" "AT1G01030" "AT1G01040" ...
  .. .. .. ..$ : chr [1:12] "scale.data.1" "scale.data.2" "scale.data.3" "scale.data.4" ...
  ..@ default   : int 1
  ..@ assay.orig: chr(0) 
  ..@ meta.data :'data.frame':	25784 obs. of  56 variables:
  .. ..$ vf_vst_counts.1_mean                 : num [1:25784] 0.1805 0.5581 0.061 0.0794 5.472 ...
  .. ..$ vf_vst_counts.1_variance             : num [1:25784] 0.4247 1.1382 0.0927 0.0859 79.2542 ...
  .. ..$ vf_vst_counts.1_variance.expected    : num [1:25784] 0.3181 1.4295 0.0943 0.1242 82.8065 ...
  .. ..$ vf_vst_counts.1_variance.standardized: num [1:25784] 1.335 0.796 0.983 0.692 0.957 ...
  .. ..$ vf_vst_counts.1_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.1_rank                 : int [1:25784] 3657 14961 7326 21005 8245 11510 1327 17728 12341 5100 ...
  .. ..$ vf_vst_counts.2_mean                 : num [1:25784] 0.1821 0.5053 0.0589 0.1071 5.1705 ...
  .. ..$ vf_vst_counts.2_variance             : num [1:25784] 0.479 1.174 0.09 0.125 92.916 ...
  .. ..$ vf_vst_counts.2_variance.expected    : num [1:25784] 0.3549 1.4576 0.0956 0.1826 99.3378 ...
  .. ..$ vf_vst_counts.2_variance.standardized: num [1:25784] 1.35 0.806 0.941 0.686 0.935 ...
  .. ..$ vf_vst_counts.2_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.2_rank                 : int [1:25784] 4060 14226 9439 19802 9520 15809 778 17671 10005 3223 ...
  .. ..$ vf_vst_counts.3_mean                 : num [1:25784] 0.2378 0.3659 0.0532 0.1083 3.5435 ...
  .. ..$ vf_vst_counts.3_variance             : num [1:25784] 0.6878 0.727 0.0909 0.1262 49.8308 ...
  .. ..$ vf_vst_counts.3_variance.expected    : num [1:25784] 0.4893 0.8968 0.0858 0.1844 49.8204 ...
  .. ..$ vf_vst_counts.3_variance.standardized: num [1:25784] 1.406 0.811 1.059 0.684 1 ...
  .. ..$ vf_vst_counts.3_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.3_rank                 : int [1:25784] 3797 14213 6191 20115 7391 14905 1369 17887 7581 2127 ...
  .. ..$ vf_vst_counts.4_mean                 : num [1:25784] 0.457 0.3227 0.0173 0.0837 5.3581 ...
  .. ..$ vf_vst_counts.4_variance             : num [1:25784] 1.6575 0.6117 0.0264 0.0979 127.7057 ...
  .. ..$ vf_vst_counts.4_variance.expected    : num [1:25784] 1.2855 0.7638 0.0245 0.1343 118.0776 ...
  .. ..$ vf_vst_counts.4_variance.standardized: num [1:25784] 1.289 0.801 1.077 0.729 1.082 ...
  .. ..$ vf_vst_counts.4_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.4_rank                 : int [1:25784] 4172 14908 5866 18827 5816 10444 658 15147 13777 1522 ...
  .. ..$ vf_vst_counts.5_mean                 : num [1:25784] 0.2857 0.2443 0.0128 0.0638 3.6844 ...
  .. ..$ vf_vst_counts.5_variance             : num [1:25784] 0.7914 0.4341 0.0171 0.0717 62.6784 ...
  .. ..$ vf_vst_counts.5_variance.expected    : num [1:25784] 0.6298 0.5074 0.0173 0.0968 58.4671 ...
  .. ..$ vf_vst_counts.5_variance.standardized: num [1:25784] 1.257 0.856 0.984 0.741 1.072 ...
  .. ..$ vf_vst_counts.5_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.5_rank                 : int [1:25784] 4044 12442 7032 18351 5637 11690 837 15413 14016 1047 ...
  .. ..$ vf_vst_counts.6_mean                 : num [1:25784] 0.5022 0.308 0.0258 0.11 6.2128 ...
  .. ..$ vf_vst_counts.6_variance             : num [1:25784] 2.7013 0.6172 0.0409 0.147 184.947 ...
  .. ..$ vf_vst_counts.6_variance.expected    : num [1:25784] 1.6137 0.7746 0.0401 0.1992 180.0732 ...
  .. ..$ vf_vst_counts.6_variance.standardized: num [1:25784] 1.674 0.797 1.021 0.738 1.027 ...
  .. ..$ vf_vst_counts.6_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.6_rank                 : int [1:25784] 2937 14753 7034 17326 6938 12513 873 17980 11059 2164 ...
  .. ..$ vf_vst_counts.7_mean                 : num [1:25784] 0.2745 0.1694 0.0183 0.067 3.4455 ...
  .. ..$ vf_vst_counts.7_variance             : num [1:25784] 0.8888 0.2388 0.0281 0.0816 53.2331 ...
  .. ..$ vf_vst_counts.7_variance.expected    : num [1:25784] 0.5999 0.3047 0.0243 0.1002 51.7187 ...
  .. ..$ vf_vst_counts.7_variance.standardized: num [1:25784] 1.482 0.784 1.156 0.815 1.029 ...
  .. ..$ vf_vst_counts.7_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.7_rank                 : int [1:25784] 2861 17807 4775 15929 6239 10828 739 15713 18815 2037 ...
  .. ..$ vf_vst_counts.8_mean                 : num [1:25784] 0.2158 0.1832 0.0125 0.074 3.3243 ...
  .. ..$ vf_vst_counts.8_variance             : num [1:25784] 0.5433 0.3145 0.0145 0.0861 70.8501 ...
  .. ..$ vf_vst_counts.8_variance.expected    : num [1:25784] 0.4455 0.3564 0.0163 0.1141 57.7609 ...
  .. ..$ vf_vst_counts.8_variance.standardized: num [1:25784] 1.22 0.883 0.886 0.755 1.227 ...
  .. ..$ vf_vst_counts.8_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.8_rank                 : int [1:25784] 4060 12412 12192 19441 4012 17866 887 16532 11411 2232 ...
  .. ..$ vf_vst_counts.9_mean                 : num [1:25784] 0.1887 0.1725 0.0167 0.0799 3.9704 ...
  .. ..$ vf_vst_counts.9_variance             : num [1:25784] 0.5524 0.2878 0.0226 0.0991 84.0922 ...
  .. ..$ vf_vst_counts.9_variance.expected    : num [1:25784] 0.388 0.343 0.023 0.129 79.812 ...
  .. ..$ vf_vst_counts.9_variance.standardized: num [1:25784] 1.425 0.839 0.985 0.767 1.054 ...
  .. ..$ vf_vst_counts.9_variable             : logi [1:25784] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. ..$ vf_vst_counts.9_rank                 : int [1:25784] 3187 14456 7487 18238 5979 16275 1351 16216 6180 2563 ...
  .. ..$ var.features                         : chr [1:25784] "AT1G01010" "AT1G01020" "AT1G01030" "AT1G01040" ...
  .. ..$ var.features.rank                    : int [1:25784] 3124 11415 5823 15658 5258 9617 664 13382 8650 1770 ...
  ..@ misc      : list()
  ..@ key       : chr "rna_"

Available layers (if any):
 [1] "data"         "counts"       "scale.data.1" "scale.data.2" "scale.data.3"
 [6] "scale.data.4" "scale.data.5" "scale.data.6" "scale.data.7" "scale.data.8"
[11] "scale.data.9" "scale.data"  

Seurat object version: 